deeppavlov/models/preprocessors/torch_transformers_preprocessor.py

```diff
@@ -31,6 +31,88 @@
 log = getLogger(__name__)
 
 
+@register('torch_transformers_multiplechoice_preprocessor')
+class TorchTransformersMultiplechoicePreprocessor(Component):
+    """Tokenize text on subtokens, encode subtokens with their indices, create tokens and segment masks.
+
+    Check details in :func:`bert_dp.preprocessing.convert_examples_to_features` function.
+
+    Args:
+        vocab_file: path to vocabulary
+        do_lower_case: set True if lowercasing is needed
+        max_seq_length: max sequence length in subtokens, including [SEP] and [CLS] tokens
+        return_tokens: whether to return tuple of input features and tokens, or only input features
+
+    Attributes:
+        max_seq_length: max sequence length in subtokens, including [SEP] and [CLS] tokens
+        return_tokens: whether to return tuple of input features and tokens, or only input features
+        tokenizer: instance of Bert FullTokenizer
+
+    """
+
+    def __init__(self,
+                 vocab_file: str,
+                 do_lower_case: bool = True,
+                 max_seq_length: int = 512,
+                 return_tokens: bool = False,
+                 **kwargs) -> None:
```

Comment:
Please, implement adding token_type_ids inside `tokenize_mc_examples`, if tokenizer does not generate it. Distil* tokenizer does not return `token_type_ids`, and this causes problems with training and evaluation. 
You should add  argument add_token_type_ids: bool = False (as done in my code) and generate vector with 0s for `text` and 1s for `text_pair` after tokenization

Reply:
The newer tokenizer does that on it's own depending on the pretrained model. For example, tokenizer from pretrained bert will return token_type_ids, while tokenizer from pretrained roberta won't. You don't have to do all of this.

Reply:
The problem might be that you're iterating over hard-coded list of features in `__call__` or `forward`.

Reply:
No, it's problem with all Distil class. It doesn't work with `token_type_ids`